cpp/ql/test/library-tests/dataflow/dataflow-tests/type-bugs.expected

```diff
@@ -1,5 +1,37 @@
 astTypeBugs
 irTypeBugs
+| ../../../include/iterator.h:21:3:21:10 | ../../../include/iterator.h:21:3:21:10 | ../../../include/iterator.h:21:3:21:10 | [summary param] *0 in iterator |
```

Comment:
I'm not totally sure what's going on with these tests. It may be related to how we model iterators in dataflow (i.e., we have special code to recognize `*it = x` as being a write to the container that created the iterator `it`). I don't think we need to fix this in this PR.

Reply:
Indeed. In fact, either the problem is that it has more than one associated type, or that it has no associated type: https://github.com/github/codeql/blob/ce177c345033595a41e9eb9931fda65d15cf887e/cpp/ql/test/library-tests/dataflow/dataflow-tests/type-bugs.ql#L21

We should add a `description` to this query predicate to make it slightly more clear what the problem is.

In any case, I plan on investigating this as a follow-up if you're okay with it. It shouldn't have any affect on query results.

Reply:
Sounds good to me.